### PR TITLE
Fix connection quality when less than 3 layers are published

### PR DESCRIPTION
**Pull request by davidzhao**

### CODE REVIEW

File: pkg/sfu/downtrack.go [hunk 1]
```diff
@@ -130,6 +130,7 @@ type DownTrack struct {
 	maxPacketTs uint32
 
 	bandwidthConstrainedMuted atomicBool
+	maxAvailableLayer         atomicInt32
```
Comment (boks1971) [inline pkg/sfu/downtrack.go:133]: Guess this needs to be ported to the latest code.

I am thinking with the latest stream allocator effort, there is going to be enough perturbance.

File: pkg/rtc/participant.go [hunk 2]
```diff
@@ -701,7 +701,13 @@ func (p *ParticipantImpl) GetConnectionQuality() livekit.ConnectionQuality {
 		if subTrack.IsMuted() {
 			continue
 		}
-		if subTrack.DownTrack().TargetSpatialLayer() < subTrack.DownTrack().MaxSpatialLayer() {
+		// determine the max spatial layer based on what's published
```
Comment (boks1971) [inline pkg/rtc/participant.go:704]: Would be good to tuck all this away in DownTrack. I believe I have some of this already there in the latest. Will look at it when doing the next version of stream allocator as that is going to shake up things a fair bit.

**Review by boks1971 — APPROVED** [review]

File: pkg/rtc/participant.go (hunk 2 above)
Comment (davidzhao) [reply pkg/rtc/participant.go:704]: sounds good.

File: pkg/sfu/downtrack.go (hunk 1 above)
Comment (davidzhao) [reply pkg/sfu/downtrack.go:133]: I checked the latest, and some form of this is already available in Forwarder. @shishirng is working on computing MOS for the next version of connection quality. and we'll work to integrate the new changes into it

**Pull request merged**